users/cacher/cacher.go

```
@@ -0,0 +1,166 @@
+	if addresses, err = rc.postgresDB.GetAllAddresses(); err != nil {
+		logger.Debugw("error from query postgres db", "error", err.Error())
+		return err
+	}
+	logger.Debugw("addresses from postgres", "addresses", addresses)
+
+	for _, address := range addresses {
+		user := common.UserResponse{
+			KYCed: true,
+		}
+		rc.saveToCache(address, user, 0)
+	}
+	return nil
+}
+
+func (rc RedisCacher) cacheRichUser() error {
+	var (
+		logger = rc.sugar.With("func", "user/cacher/cachedUserInfo")
+	)
+	// read total trade 24h
+	query := fmt.Sprintf(`SELECT SUM(amount) as daily_fiat_amount FROM 
+	(SELECT eth_amount*eth_usd_rate as amount FROM trades WHERE time <= now() AND time >= (now()-24h) GROUP BY user_addr)
+	GROUP BY user_addr`)
+
+	logger.Debugw("query", "query 24h trades", query)
+
+	res, err := influxQueryDB(rc.influxDBClient, query)
+	if err != nil {
+		logger.Debugw("error from query", "err", err)
+		return err
+	}
+
+	// loop all user, check kyced
+	if len(res) == 0 || len(res[0].Series) == 0 || len(res[0].Series[0].Values) == 0 || len(res[0].Series[0].Values[0]) < 2 {
+		logger.Debugw("influx db is empty", "result", res)
+		return nil
+	}
+	kycedCap := common.NewUserCap(true)
+	nonKycedCap := common.NewUserCap(false)
+
+	for _, serie := range res[0].Series {
+		userAddress := serie.Tags["user_address"]
+		// check kyced
+		kyced, err := rc.isKyced(userAddress)
+		if err != nil {
+			return err
+		}
+
+		// check rich
+		userTradeAmount, err := influxdb.GetFloat64FromInterface(serie.Values[0][1])
+		if err != nil {
+			logger.Debugw("values second should be a float", "value", serie.Values[0][1])
+			return nil
+		}
+
+		if (kyced && userTradeAmount < kycedCap.DailyLimit) || (!kyced && userTradeAmount < nonKycedCap.DailyLimit) {
+			// if user is not rich then it is already cached before
+			continue
+		}
+		user := common.UserResponse{
+			Rich:  true,
+			KYCed: kyced,
+		}
+
+		// save to cache with 1 hour
+		rc.saveToCache(userAddress, user, expireTime)
```

the error is ignored.